code/modules/reagents/chemistry/reagents/alcohol_reagents.dm

```diff
@@ -2009,16 +2009,25 @@ All effects don't start immediately, but rather get worse over time; the rate is
 
 /datum/reagent/consumable/ethanol/between_the_sheets/on_mob_life(mob/living/drinker, delta_time, times_fired)
 	..()
-	if(drinker.IsSleeping())
-		if(drinker.getBruteLoss() && drinker.getFireLoss()) //If you are damaged by both types, slightly increased healing but it only heals one. The more the merrier wink wink.
-			if(prob(50))
-				drinker.adjustBruteLoss(-0.25 * REM * delta_time)
-			else
-				drinker.adjustFireLoss(-0.25 * REM * delta_time)
-		else if(drinker.getBruteLoss()) //If you have only one, it still heals but not as well.
-			drinker.adjustBruteLoss(-0.2 * REM * delta_time)
-		else if(drinker.getFireLoss())
-			drinker.adjustFireLoss(-0.2 * REM * delta_time)
+	var/is_between_the_sheets = FALSE
+	for(var/obj/item/bedsheet/bedsheet in range(drinker.loc, 0))
+		if(bedsheet.loc != drinker.loc) // bedsheets in your backpack/neck don't count
+			continue
+		is_between_the_sheets = TRUE
+		break
```

Comment:
I reused the code from #65713

https://github.com/tgstation/tgstation/blob/750db433ef0ca34e432cc6132751844874994340/code/datums/status_effects/debuffs/debuffs.dm#L190-L212

I'm not sure if `locate()` also checks the bedsheets in the backpack/neck.  If it ignores the players backpack/neck then it should be used, otherwise we should use the for loop.  This is why the original code loops through all available bedsheets on the tile, including ones the player is carrying, to make sure we don't miss the check.

Reply:
Will it search the contents of the player in the turf?  I'm assuming this is why the other code avoided using it.